src/itensor.jl

```diff
@@ -245,7 +235,7 @@ ITensor(is::Indices) = ITensor(Float64, is)
 ITensor(inds::Index...) = ITensor(Float64, IndexSet(inds...))
 
 # To fix ambiguity with QN Index version
-ITensor() = ITensor(Float64, IndexSet())
+ITensor() = emptyITensor()
```

Comment:
I discussed with Matt and we agreed to hold off on this change for now, just to make users have to be explicit about getting the emptyITensor behavior. This may be changed in a future PR though.

So could you please revert this line to `ITensor() = ITensor(Float64, IndexSet())` ? (Assuming that code still works within this PR.)

Reply:
It doesn't, that's why I changed it as I explained above.

Reply:
The problem is that doing this breaks the check in the ITensor addition methods since the storage type of the ITensor is *not* `Empty` if you construct it with previous method.

Reply:
So, we could either undo this change and find a new way to implement the behaviour for `A::ITensor+B::ITensor` (maybe check there is nothing in the storage?) or keep this and the `store(A) isa NDTensors.Empty` behaviour.

Reply:
For now, we've decided that we are keeping the behavior of `ITensor()` the same, so we still want it to error as it does now:
```julia
julia> using ITensors

julia> i = Index(2)
(dim=2|id=653)

julia> ITensor() + ITensor(i)
ERROR: cannot add ITensors with different numbers of indices
[...]
```

Reply:
Currently it is defined as `ITensor() = ITensor(0)`:
```julia
julia> ITensor() == ITensor(0)
true
```
so it should act like `ITensor(0)` for the time being.

Reply:
OK, this has been restored again. It would have been nice to break out the totally speculative discussion of `emptyITensor` to an issue rather than implying I needed to add it here.

Reply:
You can also check with us before making changes in the future if you're not sure about the desired design.